docs/Templates.md

```diff
@@ -1,44 +1,262 @@
+h_tmpl["item_key_HB"] = "Text 3B"
+...
+h_tmpl.render() # add header items to third page
+
+# footer for the third page
+f_tmpl["item_key_FC"] = "Text 3C"
+f_tmpl["item_key_FD"] = "Text 3D"
+...
+f_tmpl.render() # add footer items to third page
+
+# other content on the third page
+pdf.rect(x, y, w, h, style=None)
+
+# possibly more pages
+pdf.next_page()
+...
+...
+
+# finally write everything to a file
+pdf.output("example.pdf")
+```
+
+Evidently, this can end up quite a bit more involved, but there are hardly any limits on how you can combine templated and non-templated content on each page. Just think of the different templates as of building blocks, like configurable rubber stamps, which you can apply in any combination on any page you like.
+
+Of course, you can just as well use a set of full-page templates, possibly differentiating between cover page, table of contents, normal content pages, and an index page, or something along those lines. 
+
+And here's how you can use a template several times on one page (and by extension, several times on several pages). When rendering with an `offsetx` and/or `offsety` argument, the contents of the template will end up in a different place on the page. A `rotate` argument will change its orientation, rotated around the origin of the template.:
+
+```python
+elements = [
+    {"name":"box", "type":"B", "x1":0, "y1":0, "x2":50, "y2":50,},
+    {"name":"d1", "type":"L", "x1":0, "y1":0, "x2":50, "y2":50,},
+    {"name":"d2", "type":"L", "x1":0, "y1":50, "x2":50, "y2":0,},
+    {"name":"label", "type":"T", "x1":0, "y1":52, "x2":50, "y2":57, "text":"Label",},
+]
+pdf = FPDF()
+pdf.add_page()
+templ = FlexTemplate(pdf, elements)
+templ["label"] = "Offset: 50 / 50 mm"
+templ.render(offsetx=50, offsety=50)
+templ["label"] = "Offset: 50 / 120 mm"
+templ.render(offsetx=50, offsety=120)
+templ["label"] = "Offset: 120 / 50 mm"
+templ.render(offsetx=120, offsety=50)
+templ["label"] = "Offset: 120 / 120 mm"
+templ.render(offsetx=120, offsety=120, rotate=30.0)
+pdf.output("example.pdf")
+```
+
+Since we're handling the properties of the FPDF() instance separately, the constructor signature of this class is much simpler:
+
+```python
+fpdf.template.FlexTemplate(pdf, elements=None)
+```
+
+It supports the same public methods as Template(), except for `add_page()`, which you will instead execute directly on the FPDF() instance. However, the render() method has a few more parameters and a bit different semantics:
+
+* `FlexTemplate.load_elements(elements)`
+  * An alternative to supplying the elements dict to the constructor.
+* `FlexTemplate.parse_csv(infile,  delimiter=",", decimal_sep=".", encoding=None)`
+  * Load a template CSV file instead of supplying a dict.
+* `FlexFlexTemplate.render(offsetx=0.0, offsety=0.0, rotate=0.0)`
+  * Renders the contents to the current page.
+ 
+Setting text values for specific template items is done by treating the class as a dict, with the name of the item as the key:
+
```

Comment:
Is a code snippet missing here?

Reply:
Showing what?
The template loading/parsing is demonstrated further down, when the template formats are explained.

Reply:
The colon character just seemed to imply a code snippet ^^